luigi/hadoop.py

```diff
@@ -398,10 +403,17 @@ def run_job(self, job):
         cmb_cmd = '{0} mrrunner.py combiner'.format(python_executable)
         red_cmd = '{0} mrrunner.py reduce'.format(python_executable)
 
-        # replace output with a temporary work directory
         output_final = job.output().path
-        output_tmp_fn = output_final + '-temp-' + datetime.datetime.now().isoformat().replace(':', '-')
-        tmp_target = luigi.hdfs.HdfsTarget(output_tmp_fn)
+        # atomic output: replace output with a temporary work directory
+        if self.end_job_with_atomic_move_dir:
+            if isinstance(job.output(), luigi.s3.S3FlagTarget):
+                raise TypeError("end_job_with_atomic_move_dir is not supported"
```

Comment:
Ah - https://github.com/spotify/luigi/issues/1092 details the issue.  It's not a blocker, more a question of responsibly blocking something that is broken.